src/components/Calendar.js

```diff
@@ -272,42 +272,54 @@ class Calendar extends PureComponent {
     );
   }
   onDragSelectionStart(date) {
-    this.setState({
-      drag: {
-        status: true,
-        range: { startDate: date, endDate: date },
-        disablePreview: true,
-      },
-    });
-  }
-  onDragSelectionEnd(date) {
-    const { updateRange, displayMode, onChange } = this.props;
-    if (displayMode === 'date' || !this.state.drag.status) {
+    const { onChange, dragSelectionEnabled } = this.props;
+
+    if (dragSelectionEnabled) {
+      this.setState({
+        drag: {
+          status: true,
+          range: { startDate: date, endDate: date },
+          disablePreview: true,
+        },
+      });
+    } else {
       onChange && onChange(date);
-      return;
     }
-    const newRange = {
-      startDate: this.state.drag.range.startDate,
-      endDate: date,
-    };
-    if (displayMode !== 'dateRange' || isSameDay(newRange.startDate, date)) {
-      this.setState({ drag: { status: false, range: {} } }, () => onChange && onChange(date));
-    } else {
-      this.setState({ drag: { status: false, range: {} } }, () => {
-        updateRange && updateRange(newRange);
-      });
+  }
+
+  onDragSelectionEnd(date) {
+    const { updateRange, displayMode, onChange, dragSelectionEnabled } = this.props;
+
+    if (dragSelectionEnabled) {
+      if (displayMode === 'date' || !this.state.drag.status) {
+        onChange && onChange(date);
+        return;
+      }
+      const newRange = {
+        startDate: this.state.drag.range.startDate,
+        endDate: date,
+      };
+      if (displayMode !== 'dateRange' || isSameDay(newRange.startDate, date)) {
+        this.setState({ drag: { status: false, range: {} } }, () => onChange && onChange(date));
+      } else {
+        this.setState({ drag: { status: false, range: {} } }, () => {
+          updateRange && updateRange(newRange);
+        });
+      }
     }
   }
   onDragSelectionMove(date) {
-    const { drag } = this.state;
-    if (!drag.status) return;
-    this.setState({
-      drag: {
-        status: drag.status,
-        range: { startDate: drag.range.startDate, endDate: date },
-        disablePreview: true,
-      },
-    });
+    if (this.props.dragSelectionEnabled) {
```

Comment:
there is no need to wrap whole func with if condition again. You can add an `or` to L314

```
if (!drag.status || !this.props.dragSelectionEnabled) return;
```